### ISSUE TITLE
Adding another classification branch, trouble using ignore_index to make new branch prediction on single COCO category while ignoring others

### ISSUE THREAD

**Issue by Feynman27** [facebookresearch/Detectron#709]
### Problem Statement
I am trying to add another classification branch (separate from the fast rcnn bbox and cls heads) to my detectron model. This branch simply consists of 2 mlps and makes predictions off of the foreground rois (similar to the keypoints branch). 

I was able to successfully train a model using only the `person` category in the fast rcnn classification head i.e. the fast rcnn classification predicted only `background` or `person`, and the additional gender classification branch predicted `background`, `male`, or `female`.  Now, I'd like to train a detection model with gender classification using all of the original COCO classes (e.g. `aeroplane`, `bicycle`, etc), but would like to ignore the gender classification for all categories except the `person` category. 

I intend to ignore these other COCO categories by simply labeling the `gender` as -1 in the COCO annotation file for these instances and calculating the gender classification loss only over the non-ignore targets. The option to ignore an index exists in the [CrossEntropyLoss](https://pytorch.org/docs/stable/nn.html#crossentropyloss) of PyTorch. However, in Caffe2, I don't think this is currently supported (see [LabelCrossEntropyKernel](https://github.com/pytorch/pytorch/blob/7035975508ed053b5f1ac08b96ac6d6b2bbb954e/caffe2/operators/softmax_ops.cu#L13)). 

I've tried altering the cuda code myself with no luck thus far. I've also made a feature request [here](https://github.com/pytorch/pytorch/issues/12576). Is modifying the Caffe2 cuda implementation of the `LabelCrossEntropyKernel` the best approach, or is there a better solution that I'm missing?
  
### Expected results

An additional loss term for the gender classification head. This loss term should be 0.0 or ignored for non-applicable classes from the fast rcnn classification head.

### Actual results

After changing the cuda implementation of Caffe2's CE loss, the loss either goes `nan` and exits or I get crashes with various errors that I don't quite understand e.g.

```
E1016 19:02:12.067680 14293 net_async_base.cc:368] [enforce fail at tensor.h:529] size_ >= 0. Tensor is not initialized. You probably need to call Resize() before calling mutable_data() Error from operator: 
input: "gpu_0/cls_score" input: "gpu_0/labels_int32" input: "gpu_0/cls_prob" input: "gpu_0/loss_cls_grad" output: "gpu_0/cls_score_grad" name: "" type: "SoftmaxWithLossGradient" arg { name: "scale" f: 1 } device_option { device_type: 1 cuda_gpu_id: 0 } is_gradient_op: true,  op SoftmaxWithLossGradient
``` 

### Detailed steps to reproduce

Create an additional classification branch in detectron, and set the target value to `-1` for all COCO classes except the `person` class, which may have values `1` or `2` (`0` is background).

### System information

* Operating system: Ubuntu 16.04
* Compiler version: Cuda compilation tools, release 9.2, V9.2.148
* CUDA version: 9020
* cuDNN version: 7104
* NVIDIA driver version: 396.37
* GPU models (for all devices if they are not all the same): Tesla V100
* `PYTHONPATH` environment variable: /home/thomas/pytorch/build
* `python --version` output: ython 2.7.15 :: Anaconda, Inc.
* Anything else that seems relevant: ?
